### PR TITLE
feat(issue-alerts): Add latest adopted release filter to project rules config endpoint

### PR DESCRIPTION
**Pull request by wedamija**
This adds the latest adopted release filter ot the project rules config endpoint, and adds tests for the form validation.

Also renamed some of the parameters of the filter so that they'd read in a more intuitive way to the user.

### CODE REVIEW

File: src/sentry/api/endpoints/project_rules_configuration.py [hunk 1]
```diff
@@ -33,6 +33,9 @@ def get(self, request: Request, project) -> Response:
             "organizations:integrations-ticket-rules", project.organization
         )
         has_issue_severity_alerts = features.has("projects:first-event-severity-alerting", project)
+        has_latest_adopted_release = features.has(
+            "organizations:latest-adopted-release-filter", project
```
Comment (scefali) [inline src/sentry/api/endpoints/project_rules_configuration.py:37]: @wedamija shouldn't you pass in `project.organization`?

File: src/sentry/rules/filters/latest_adopted_release_filter.py [hunk 2]
```diff
@@ -185,8 +187,8 @@ def clear_get_first_last_release_for_group_cache(instance: GroupRelease, **kwarg
     )
 
 
-def get_project_release_cache_key(project_id: int, environment_id: int) -> str:
-    return f"project:{project_id}:env:{environment_id}:latest_release_adopted"
+def get_project_release_cache_key(project_id: int, environment: str) -> str:
+    return f"project:{project_id}:env:{environment}:latest_release_adopted"
```
Comment (scefali) [inline src/sentry/rules/filters/latest_adopted_release_filter.py:191]: @wedamija why are we changing the cache key to the name from the id?

File: src/sentry/api/endpoints/project_rules_configuration.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/api/endpoints/project_rules_configuration.py:37]: Oof, that's what I get for copy/pasting feature logic...

File: src/sentry/rules/filters/latest_adopted_release_filter.py [hunk 3]
```diff
@@ -28,57 +29,58 @@
 
 
 class LatestAdoptedReleaseForm(forms.Form):
-    release_age_type = forms.ChoiceField(choices=list(model_age_choices))
-    age_comparison = forms.ChoiceField(choices=list(age_comparison_choices))
-    environment_id = forms.CharField(required=False)
+    oldest_or_newest = forms.ChoiceField(choices=list(model_age_choices))
+    older_or_newer = forms.ChoiceField(choices=list(age_comparison_choices))
+    environment = forms.CharField(required=False)
```
Comment (vartec) [inline src/sentry/rules/filters/latest_adopted_release_filter.py:34]: Not a blocker: having env as a ChoiceField pre-populated with org's valid envs would be nice.

File: src/sentry/rules/filters/latest_adopted_release_filter.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/rules/filters/latest_adopted_release_filter.py:191]: I changed the parameter to accept the env name for consistency with other related endpoints and so just changed this as well. It's breaking tests though, so I'll rework this.

**Review by vartec — APPROVED** [review]
lgtm

File: src/sentry/rules/filters/latest_adopted_release_filter.py (hunk 3 above)
Comment (vartec) [inline src/sentry/rules/filters/latest_adopted_release_filter.py:34]: it's marked as`required=False`, but I don't see any default. Wouldn't not setting it just default to `""` and then raise  `forms.ValidationError( "environment does not exist...`?

Comment (wedamija) [reply src/sentry/rules/filters/latest_adopted_release_filter.py:34]: Not practical unfortunately - there are orgs with 100ks of envs

Comment (wedamija) [reply src/sentry/rules/filters/latest_adopted_release_filter.py:34]: Ahh it's required, just an oversight there

**Review by scefali — APPROVED** [review]

**Pull request merged**